girder/queues/queues/models/queue.py

```diff
@@ -246,24 +257,27 @@ def _start_taskflow(self, taskflow_id, params, user):
 
         return workflow
 
-def taskflow_status_callback(taskflow_id, queue, user):
-
-    def callback(event):
-        taskflow = event.info['taskflow']
-
-        if taskflow['_id'] != taskflow_id:
-            return
-
-        non_running_states = [
-            TaskFlowState.COMPLETE,
-            TaskFlowState.ERROR,
-            TaskFlowState.UNEXPECTEDERROR,
-            TaskFlowState.TERMINATED,
-            TaskFlowState.DELETED
-        ]
-
-        if taskflow['status'] in non_running_states:
-            Queue().finish(queue, taskflow, user)
-            Queue().pop(queue, sys.maxsize, user)
-
-    return callback
+def cleanup_failed_taskflows():
+    queues = list(Queue().find(limit=sys.maxsize, force=True))
+    for queue in queues:
+        user = UserModel().load(queue['userId'], force=True)
+        if user is None:
+            continue
+
+        for taskflow_id, status in queue['taskflows'].items():
+            if status == TaskStatus.RUNNING:
+                taskflow = TaskflowModel().load(taskflow_id, force=True)
+                if taskflow['status'] in TASKFLOW_NON_RUNNING_STATES:
+                    Queue().finish(queue, taskflow, user)
```

Comment:
Can we add a `logger.warn` for these tasks being removed from the queue.

Reply:
Done!